test/system/scenarios/test_electrodes.py

```diff
@@ -426,6 +426,119 @@ def issue497(sim):
     assert_true (abs(i_amp_ac2[int(start2/sim_dt)]-amplitude) < 1e-9)
 
 
+@register()
+def issue512(sim):
+    """
+    Test to ensure that StepCurrentSource times are handled similarly across
+    all simulators. Multiple combinations of step times tested for:
+    1) dt = 0.1 ms, min_delay = 0.1 ms
+    2) dt = 0.01 ms, min_delay = 0.01 ms
+    Note: exact matches of times not appropriate owing to floating point
+    rounding errors. If absolute difference <1e-9, then considered equal.
+    """
+    def get_len(data):
+        if "pyNN.nest" in str(sim):
+            # as NEST uses LazyArray
+            return len(data.evaluate())
+        else:
```

Comment:
don't pyNN.neuron and pyNN.brian also use lazyarray?

Reply:
if not, they should

Reply:
I believe they do not currently. Because I vaguely recollect getting a LazyArray related error message for NEST, while not for the other two. Even the above code, which runs successfully for all three simulators seems to suggest so.

Could we create a separate issue to investigate/fix this?

Reply:
yes

Reply:
Created #542